### PR TITLE
Integrate simdjson with VAST

### PR DESCRIPTION
**Pull request by dominiklohmann**
###  :notebook_with_decorative_cover: Description

This PR serves as a merge for base @ngrodzitski's work on integrating simdjson with VAST.

All CI changes have to take place in this PR, because GitHub does not use Workflow changes from forks.

###  :memo: Checklist

- [x] All user-facing changes have changelog entries.
- [x] The changes are reflected on [docs.tenzir.com/vast](https://docs.tenzir.com/vast), if necessary.
- [x] The PR description contains instructions for the reviewer, if necessary.

### :dart: Review Instructions

n/t

### CODE REVIEW

**Comment by dominiklohmann** [comment]
[ch4981]

File: .github/workflows/vast.yaml [hunk 1]
```diff
@@ -98,14 +98,23 @@ jobs:
         if: matrix.os.tag == 'Ubuntu'
         run: |
           sudo apt-get -qq update
-          sudo apt-get -qqy install ninja-build libpcap-dev libssl-dev lsb-release ccache libflatbuffers-dev libyaml-cpp-dev
+          sudo apt-get -qqy install ninja-build libpcap-dev libssl-dev lsb-release ccache libflatbuffers-dev libyaml-cpp-dev libsimdjson-dev
```
Comment (dominiklohmann) [inline .github/workflows/vast.yaml:101]: Remove simdjson here

File: libvast/vast/format/json/default_selector.hpp [hunk 2]
```diff
@@ -28,9 +28,39 @@
...
+      }
+      prefix.pop_back();
+    }
+  }
+
+  static auto make_names_layout(const ::simdjson::dom::object& obj) {
+    std::vector<std::string> entries;
+    entries.reserve(100);
+    auto prefix = detail::stack_vector<std::string_view, 64>{};
+
```
Comment (dominiklohmann) [inline libvast/vast/format/json/default_selector.hpp:56]: Remove whitespace

File: libvast/vast/format/json/default_selector.hpp [hunk 3]
```diff
@@ -50,6 +80,21 @@ struct default_selector {
...
 
+  caf::optional<record_type>
+  operator()(const ::simdjson::dom::object& obj) const {
+    if (type_cache.empty())
+      return caf::none;
+    // Iff there is only one type in the type cache, allow the JSON reader to
+    // use it despite not being an exact match.
+    if (type_cache.size() == 1)
+      return type_cache.begin()->second;
+
```
Comment (dominiklohmann) [inline libvast/vast/format/json/default_selector.hpp:91]: Remove whitespace

File: libvast/vast/format/json/field_selector.hpp [hunk 4]
```diff
@@ -51,6 +52,27 @@ struct field_selector {
     return type;
   }
 
+  caf::optional<vast::record_type>
+  operator()(const ::simdjson::dom::object& j) {
+    auto el = j.at_key(Specification::field);
+    if (el.error())
+      return caf::none;
+
```
Comment (dominiklohmann) [inline libvast/vast/format/json/field_selector.hpp:60]: Remove whitespace

File: libvast/vast/format/simdjson.hpp [hunk 5]
```diff
@@ -0,0 +1,209 @@
...
+#include "vast/logger.hpp"
+#include "vast/schema.hpp"
+#include "vast/view.hpp"
+
+#include <caf/expected.hpp>
+#include <caf/fwd.hpp>
+#include <caf/settings.hpp>
+
+#include <chrono>
+#include <simdjson.h>
```
Comment (dominiklohmann) [inline libvast/vast/format/simdjson.hpp:36]: Make public in CMake

File: libvast/src/format/simdjson.cpp [hunk 6]
```diff
@@ -0,0 +1,396 @@
...
+}
+
+template <>
+caf::expected<data> type_biased_convert_impl<::simdjson::dom::object, map>(
+  ::simdjson::dom::object o, const type& t) {
+  const auto& m = dynamic_cast<const map_type&>(*t);
+  map xs;
+  xs.reserve(o.size());
+  for (auto [k, v] : o) {
+    // TODO: Properly unwrap the key type instead of wrapping it in json.
```
Comment (dominiklohmann) [inline libvast/src/format/simdjson.cpp:259]:
```suggestion
    // TODO: Properly unwrap the key type instead of wrapping it in JSON.
```

File: CHANGELOG.md [hunk 7]
```diff
@@ -12,6 +12,14 @@ Every entry has a category for which we use the following visual abbreviations:
...
 ## Unreleased
 
+- 🧬 As a first step to increasing the performance of our JSON import, VAST now
+  relies on [simdjson](https://github.com/simdjson/simdjson) to parse NDJSON.
+  To use the (yet experimental) feature, use `vast import json|suricata
+  --simdjson`. Special thanks to [@ngrodzitski](https://github.com/ngrodzitski)
+  for all the work on this integration.
+  [#1230](https://github.com/tenzir/vast/pull/1230)
+  [#1246](https://github.com/tenzir/vast/pull/1246)
+
```
Comment (mavam) [inline CHANGELOG.md:22]:
```suggestion
- 🧬 VAST now relies on [simdjson](https://github.com/simdjson/simdjson) for
  JSON parsing. The substantial gains in throughput shift the bottleneck of
  the ingest path from parsing input to indexing at the node. To use the (yet
  experimental) feature, use `vast import json|suricata --simdjson`.
  [#1230](https://github.com/tenzir/vast/pull/1230)
  [#1246](https://github.com/tenzir/vast/pull/1246)
  [@ngrodzitski](https://github.com/ngrodzitski)
```

We are going to acknowledge this external contribution separately in the README. Let's do this in a different PR.

**Review by mavam — APPROVED** [review]
Super excited to see this landing in master!
